gems/aws-crt/spec/auth/static_credentials_provider_spec.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+require_relative '../spec_helper'
+require 'weakref'
+
+module Aws
+  module Crt
+    module Auth #:nodoc:
+      describe StaticCredentialsProvider do
+        describe '#initilize' do
+          it 'raises an ArgumentError when missing access_key_id' do
+            expect { StaticCredentialsProvider.new(nil, 'secret') }
+              .to raise_error(ArgumentError)
+          end
+
+          it 'raises an ArgumentError when missing secret_access_key' do
+            expect { StaticCredentialsProvider.new('akid', nil) }
+              .to raise_error(ArgumentError)
+          end
+        end
+
+        describe '.on_release' do
+          it 'cleans up with release' do
+            creds = StaticCredentialsProvider.new('akid', 'secret')
+            expect(creds).to_not be_nil
+
+            creds.release
+            check_for_clean_shutdown
+          end
+
+          if garbage_collect_is_immediate?
```

Comment:
Should this be some sort of shared RSpec example for classes that implement Native? Or perhaps this shouldn't be tested because the library is supposed to do it?

Reply:
I'm not sure how to do this as a shared example - each binding needs to construct its own object and these tests do not work if `let` is used.  But yes - ideally these would have been a shared example.